### PR TITLE
test(e2e): add e2e tests for message-toast component

### PR DESCRIPTION
**Pull request by maxym-valor**
#### Please provide a link to the associated issue.
Closes

#### Please check whether the PR fulfills the following requirements

- [ ] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [ ] tests for the changes that have been done
- [ ] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [ ] Run npm run build-pack-library and test in external application

Documentation checklist:
- [ ] update `README.md`
- [ ] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [ ] Documentation Examples
- [ ] Stackblitz works for all examples

### CODE REVIEW

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    beforeEach(() => {
+        refreshPage();
+        waitForElDisplayed(messageToastPage.title);
+    }, 2);
+
+    it('should check message from component', () => {
+        click(openMessageButton);
+        expect(isElementDisplayed(message)).toBe(true, 'message is not displayed');
+        pause(7500);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:34]: I have a suggestion.. either do waitForElNoDisplayed(element, index, 7500); if you want to check it's gone in the correct time
or click the 'hide all' btn if you don't want to check the timing.

pause should only be used if absolutely needed, i.e. animation like column transition etc.

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        expect(doesItExist(message)).toBe(false, 'message still displayed');
+        click(openMessageButton);
+        mouseHoverElement(message);
+        pause(10000);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:38]: same comment as above

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        expect(isElementDisplayed(message)).toBe(true, 'message is not displayed');
+    })
+
+    it('should check message from template', () => {
+        scrollIntoView(openMessageButton, 1);
+        click(openMessageButton, 1);
+        expect(isElementDisplayed(message)).toBe(true, 'message is not displayed');
+        pause(4000);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:46]: same comment as above

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        expect(doesItExist(message)).toBe(false, 'message still displayed');
+    })
+
+    it('should check message from string', () => {
+        click(openMessageButton, 2);
+        expect(isElementDisplayed(message)).toBe(true, 'message is not displayed');
+        pause(5000);
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:53]: same comment as above

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    it('should check working of Hide All button', () => {
+        for (let i = 2; i >= 0; i--) {
+            click(openMessageButton, i);
+        }
+        expect(getElementArrayLength(message)).toBe(3, 'not all messages displayed');
+        scrollIntoView(hideAllButton);
+        click(hideAllButton);
+        expect(getElementArrayLength(message)).toBe(0, 'elements is not hidden');
+        expect(doesItExist(message)).toBe(false, 'message still displayed');
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:71]: I think these 2 assertions are checking the same exact thing, only 1 is needed.

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    beforeAll(() => {
+        messageToastPage.open();
+    }, 1);
+
+    beforeEach(() => {
+        refreshPage();
+        waitForElDisplayed(messageToastPage.title);
+    }, 2);
+
+    it('should check message from component', () => {
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:31]: should check what? please make these more descriptive

'should check the component example message popup appears' or ' should check component example message appears on click'

same for the other test cases below

File: e2e/wdio/core/pages/message-toast.po.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    saveExampleBaselineScreenshot(specName: string = 'message-toast'): void {
+        super.saveExampleBaselineScreenshot(specName, this.getScreenshotFolder());
+    }
+
+    compareWithBaseline(specName: string = 'message-toast'): any {
+        return super.compareWithBaseline(specName, this.getScreenshotFolder());
+    }
+}
```
Comment (Seamoo13) [inline e2e/wdio/core/pages/message-toast.po.ts:30]: always end files with new line

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    it('should check working of Hide All button', () => {
+        for (let i = 2; i >= 0; i--) {
+            click(openMessageButton, i);
+        }
+        expect(getElementArrayLength(message)).toBe(3, 'not all messages displayed');
+        scrollIntoView(hideAllButton);
+        click(hideAllButton);
+        expect(getElementArrayLength(message)).toBe(0, 'elements is not hidden');
+        expect(doesItExist(message)).toBe(false, 'message still displayed');
+    })
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:72]: `;`

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+        expect(isElementDisplayed(message)).toBe(true, 'message is not displayed');
+    })
+
+    it('should check message from template', () => {
+        scrollIntoView(openMessageButton, 1);
+        click(openMessageButton, 1);
+        expect(isElementDisplayed(message)).toBe(true, 'message is not displayed');
+        pause(4000);
+        expect(doesItExist(message)).toBe(false, 'message still displayed');
+    })
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:48]: `;`

please check the file for this

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+
+    it('should check orientation', () => {
+        messageToastPage.checkRtlSwitch();
+    });
+
+    it('should check visual regression for all examples', () => {
+        messageToastPage.saveExampleBaselineScreenshot();
+        expect(messageToastPage.compareWithBaseline()).toBeLessThan(5);
+    });
+
```
Comment (Seamoo13) [inline e2e/wdio/core/tests/message-toast.e2e-spec.ts:82]: can add one more test case:
- check component example message stays open while mouse hover message

File: e2e/wdio/core/pages/message-toast.po.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import { CoreBaseComponentPo } from './core-base-component.po';
+import { waitForElDisplayed, waitForPresent } from '../../driver/wdio';
+
+export class MessageToastPo extends CoreBaseComponentPo {
+    private url = '/message-toast';
+
+    section = 'fd-message-toast-example ';
```
Comment (Seamoo13) [inline e2e/wdio/core/pages/message-toast.po.ts:7]: let's make all the name descriptive. 'section' is understood by you and me, but maybe not to someone else 6 months from now. let's try: exampleBlock or something like that

File: e2e/wdio/core/pages/message-toast.po.ts [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    openMessageButton = this.section + '.fd-button';
+    hideAllButton = '.fd-button--emphasized';
+    messageContainer = '.fd-message-toast-container ';
+    message = '.fd-message-toast';
```
Comment (Seamoo13) [inline e2e/wdio/core/pages/message-toast.po.ts:11]: maybe messageToast, because message sounds like some text

**Review by Seamoo13 — CHANGES_REQUESTED** [review]

File: e2e/wdio/core/tests/message-toast.e2e-spec.ts (hunk 10 above)
Comment (maxym-valor) [reply e2e/wdio/core/tests/message-toast.e2e-spec.ts:82]: i did it, line 37

**Comment by Seamoo13** [comment]
raised on main repo

**Pull request closed without merging**